example/tutorial/train_diabetes.py

```diff
@@ -0,0 +1,161 @@
+#
+# train_diabetes.py
+#
+#   Modified version of train.py from Wine Quality dataset
+#       https://github.com/databricks/mlflow/blob/master/example/tutorial/train.py
+#       The data set used in this example is from http://archive.ics.uci.edu/ml/datasets/Wine+Quality
+#       P. Cortez, A. Cerdeira, F. Almeida, T. Matos and J. Reis.
+#       Modeling wine preferences by data mining from physicochemical properties. In Decision Support Systems, Elsevier, 47(4):547-553, 2009.
+#   Combines the above with the Lasso Descent Path Plot
+#       http://scikit-learn.org/stable/auto_examples/linear_model/plot_lasso_coordinate_descent_path.html
+#       Original author: Alexandre Gramfort <alexandre.gramfort@inria.fr>; License: BSD 3 clause
+#
+
+import os
+import warnings
+import sys
+
+import pandas as pd
+import numpy as np
+from itertools import cycle
+import matplotlib.pyplot as plt
+from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score
+from sklearn.model_selection import train_test_split
+from sklearn.linear_model import ElasticNet
+from sklearn.linear_model import lasso_path, enet_path
+from sklearn import datasets
+
+# Load Diabetes datasets
+diabetes = datasets.load_diabetes()
+X = diabetes.data
+y = diabetes.target
+
+# Create pandas DataFrame for sklearn ElasticNet linear_model
+Y = np.array([y]).transpose()
+d = np.concatenate((X, Y), axis=1)
+cols = diabetes.feature_names + ['progression']
+data = pd.DataFrame(d, columns=cols)
+
+
+# Import mlflow
+import mlflow
+import mlflow.sklearn
+
+
+# Evaluate metrics
+def eval_metrics(actual, pred):
+    rmse = np.sqrt(mean_squared_error(actual, pred))
+    mae = mean_absolute_error(actual, pred)
+    r2 = r2_score(actual, pred)
+    return rmse, mae, r2
+
+
+
+if __name__ == "__main__":
+    warnings.filterwarnings("ignore")
+    np.random.seed(40)
+
+    # Split the data into training and test sets. (0.75, 0.25) split.
+    train, test = train_test_split(data)
+
+    # The predicted column is "progression" which is a quantitative measure of disease progression one year after baseline
+    train_x = train.drop(["progression"], axis=1)
+    test_x = test.drop(["progression"], axis=1)
+    train_y = train[["progression"]]
+    test_y = test[["progression"]]
+
+    alpha = float(sys.argv[1]) if len(sys.argv) > 1 else 0.05
+    l1_ratio = float(sys.argv[2]) if len(sys.argv) > 2 else 0.05
+
+    with mlflow.start_run():
```

Comment:
I didn't get why you have to use "with" here since this application only does one run. It might be better to not call it at all to make this example easier to understand. The "with" is useful in notebooks or in applications that do their own parameter search and want to report multiple runs.

Reply:
Ahh good point, updated and removed the`with mlflow.start_run()` call.  Thanks!